waiter/src/waiter/auth/oidc.clj

```diff
@@ -206,29 +213,35 @@
 
 (defn make-oidc-auth-response-updater
   "Returns a response updater that rewrites 401 waiter responses to 302 redirects."
-  [jwt-auth-server password request]
+  [jwt-auth-server oidc-mode password request]
   (fn update-oidc-auth-response [{:keys [status] :as response}]
     (if (and (= status http-401-unauthorized)
              (utils/waiter-generated-response? response))
       ;; issue 302 redirect
-      (trigger-authorize-redirect jwt-auth-server password request response)
+      (trigger-authorize-redirect jwt-auth-server oidc-mode password request response)
       ;; non-401 response, avoid authentication challenge
       response)))
 
-(defn oidc-enabled-on-service?
-  "Returns true if OIDC auth is enabled for the service.
-   Result depends on whether OIDC auth is configured explicitly based on env variable or computed using allow-oidc-auth-services?"
-  [allow-oidc-auth-services? request]
+(defn retrieve-oidc-mode-on-service
+  "Returns the OIDC mode, one of :disabled :relaxed or :strict, if OIDC auth is enabled for the service.
+   Result depends on whether OIDC auth is configured explicitly based on env variable or
+   computed using allow-oidc-auth-services? and oidc-default-mode."
+  [allow-oidc-auth-services? oidc-default-mode request]
   ;; service requests will enable OIDC auth based on env variable or when absent, allow-oidc-auth-services?
   (let [use-oidc-auth-env (get-in request [:waiter-discovery :service-description-template "env" "USE_OIDC_AUTH"])]
-    (if (some? use-oidc-auth-env) (= "true" use-oidc-auth-env) allow-oidc-auth-services?)))
+    (if (some? use-oidc-auth-env)
```

Comment:
For the config/defaults, you use two parameters to control behavior: `allow-oidc-auth-services?` and `oidc-default-mode`.  For services, you use one.  They should mirror each other in function.  Either 2 for both or 1 for both.

Reply:
Do you mind rephrasing the concern?